### PR TITLE
Fix failing tests due to updated `r2dii.data` data

### LINKED ISSUE #434: Figure out why 9 tests fail
This PR is the culprit: 
https://github.com/RMI-PACTA/r2dii.match/commit/7a2b5921e933ec3dc4c3a2470ac20ee6cd51f37a

The call to `skip_on_cran()` there seems to skip about 87 tests. 

Without `skip()` (179 total tests, 9 failing):
Screenshot 2024-01-10 at 12 34 07

with `skip()` (92 total tests, all passing):
Screenshot 2024-01-10 at 12 35 29

Those tests aren't skipped on CI and are why this PR is failing. 

I will look into why all those tests are failing in a separate issue...

_Originally posted by @jdhoffa in https://github.com/RMI-PACTA/r2dii.match/issues/433#issuecomment-1885409281_

### PR DESCRIPTION
**Pull request by jdhoffa**
I did a handful of things to ensure the tests are passing as is, mainly to do with adapting tests to the updated format of `loanbook_demo`:
- update snapshots
- update expected col order
- adapt test to new comp names
- point to relevant entry

I also removed the call to  `skip_on_cran`, as this should work fine now. 

No update to `NEWS.md` required, as this is not user-facing. 

Relates to #432
Closes #434

### CODE REVIEW

**Comment by jdhoffa** [comment]
FYI. @AlexAxthelm

File: tests/testthat/test-match_name.R [hunk 1]
```diff
@@ -268,7 +268,7 @@ test_that("works with `min_score = 0` (bug fix)", {
 })
 
 test_that("outputs only perfect matches if any (#40 @2diiKlaus)", {
-  this_name <- "large hdv company three"
+  this_name <- "Jacob"
```
Comment (jacobvjk) [inline tests/testthat/test-match_name.R:271]: lol

**Review by jacobvjk — COMMENTED** [review]
@jdhoffa any idea why a test is failing for some windows checks? the errors seem to point to an issue that we thought was closed: https://github.com/RMI-PACTA/r2dii.match/issues/425

File: tests/testthat/test-match_name.R (hunk 1 above)
Comment (jdhoffa) [reply tests/testthat/test-match_name.R:271]: AHahah it is genuinely a fake company name in the updated data @cjyetman made, ¯\_(ツ)_/¯

**Comment by jdhoffa** [comment]
> @jdhoffa any idea why a test is failing for some windows checks? the errors seem to point to an issue that we thought was closed: #425

Hmmm, I do not know why this test is failing... I would say this is not related directly to this PR (updating regressions from `r2dii.data`), and I would elect to open a new issue to handle this persistent Windows encoding issue.

**Review by jacobvjk — APPROVED** [review]
lgtm

**Pull request merged**